### ISSUE TITLE
Bug: Complex chain AD (einsum + scalar op) produces incorrect gradients

### ISSUE THREAD

**Issue by shinaoka** [tensor4all/tenferro-rs#556]
## Description

When chaining einsum with element-wise scalar operations (e.g., `exp`) on **Complex64 tensors**, both VJP and JVP produce incorrect gradients that do not match central finite differences.

## Reproduction

```rust
use num_complex::Complex64;
use tenferro::{forward_ad, grad, set_default_runtime, GradOptions, RuntimeContext, Tensor};
use tenferro_prims::CpuContext;

let _g = set_default_runtime(RuntimeContext::Cpu(CpuContext::new(1)));
let a_data = vec![
    Complex64::new(0.1, 0.05), Complex64::new(0.2, -0.03),
    Complex64::new(0.15, 0.02), Complex64::new(0.3, -0.01),
];
let b_data = vec![
    Complex64::new(0.2, 0.01), Complex64::new(0.05, -0.02),
    Complex64::new(-0.1, 0.07), Complex64::new(0.15, 0.03),
];

// VJP
let mut a = Tensor::from_slice(&a_data, &[2, 2]).unwrap();
let b = Tensor::from_slice(&b_data, &[2, 2]).unwrap();
a.set_requires_grad(true).unwrap();
let c = Tensor::einsum("ij,jk->ik", &[&a, &b]).unwrap();
let e = c.exp().unwrap();
let loss = e.sum().unwrap();
let grads = grad(&[&loss], &[&a], None, GradOptions::default()).unwrap();
// Gradient does NOT match finite differences: error ~3e-4 (should be ~1e-8)
```

## Observed behavior

- VJP directional error: ~3e-4 (expected < 1e-8)
- JVP element-wise error: ~1.9e-4 (expected < 1e-8)
- **Real tensors with the same chain work correctly** (error ~5e-10)
- **Complex einsum-only and complex exp-only VJP/JVP are correct individually**

## Likely root cause

Convention mismatch between:
- `einsum_rrule` / `einsum_frule` (non-conjugated complex adjoint)  
- Scalar rrule/frule in `chainrules` (e.g., `exp_rrule` uses `cotangent * conj(result)` — Wirtinger conjugate adjoint)

When these two rules are composed in a chain, the cotangent passed between them carries the wrong semantic.

## Affected operations

Any computation graph mixing einsum with element-wise scalar operations (`exp`, `log`, `sin`, `cos`, `sqrt`, etc.) on Complex64/Complex32 tensors.
